test/test_email_reply_parser.py

```diff
@@ -248,6 +248,13 @@ def test_include_signature_false_french(self):
         body = EmailReplyParser.cut_off_at_signature(message.text, include=False, word_limit=100)
         assert body.endswith("s'il vous plait?")
 
+    def test_clean_email_portuguese(self):
+        # Similar to previous test except for French email
+        message = self.get_email('email_portuguese')
+        body = EmailReplyParser.cut_off_at_signature(message.text, word_limit=100)
+        print(body)
+        assert body.endswith("Cumprimentos Pedro Mota")
+
```

Comment:
Just out of curiosity, why does the parser merge this all onto one line?

Reply:
OK so the issue starts in the `def keep_signoff(` function where if no signoff is found we trim to the word limit (see `def word_limit_cut_off(`). 

I think to fix this you just need to modify that function to:
```python
@staticmethod
  def word_limit_cut_off(body, word_limit):
      """
      Cut-off email at word limit.
      """
      if word_limit is not None:
          # if we can't find the sign-off, let's just take the first 100 words
          email_body_list = body.replace("\n", " \n").split(" ")
          newlines_offset = len([word for word in email_body_list if word == " \n"])
          body = " ".join(email_body_list[:word_limit + newlines_offset]).replace(" \n", "\n")

      return body
```
Without a default the split function will look for any white space including a newline so we will just specify the " " literally then handle any cases where we have just a plain newline by adding an offset onto the end.

I'd say you'll have to check the various tests on the watcher then once we merge this